app/services/submission/trait_score_parser.rb

```diff
@@ -34,16 +34,20 @@ def call
 
   def map_headers_to_traits
     @upload.log "Mapping file header columns to Trait Descriptors"
-    header = input.readline
+    header = csv.readline
     @trait_names = PlantTrialSubmissionDecorator.decorate(@upload.submission).sorted_trait_names
     @trait_mapping = {}
-    header.split("\t")[1..-1].each_with_index do |column_name, i|
-      next if i >= @trait_names.length
-      column_name = column_name.strip
-      trait_index = @trait_names.find_index(column_name)
-      trait_index = i unless trait_index.present?
-      @trait_mapping[i] = trait_index
-      @upload.log " - Mapped column '#{column_name}' to Trait index #{trait_index} of value #{@trait_names[trait_index]}"
+    if header.blank?
```

Comment:
You are right but that's exactly why I did this. Paperclip's content spoofing detection is so faulty, I was recently forced to disable it in the other project (it completely confused one file type with another). I have a feeling that sooner or later we'll disable it here as well :(. Then, this will help us.

First line is not required to start with 'Plant scoring unit name' - our template suggests it but one is able to use own's template (and then the first column header is simply ignored).